packages/core/src/header-builder/csrf-token-header.ts

```diff
@@ -49,12 +50,30 @@ function makeCsrfRequest<T extends HttpRequestConfig>(
     .then(response => response.headers)
     .catch(error => {
       if (!error.response) {
+        // TODO: remove once https://github.com/axios/axios/issues/3369 is fixed
+        const retry = axiosWorkaround(error, requestConfig, destination);
+        if (retry) {
+          return retry;
+        }
         throw errorWithCause('The error response is undefined.', error);
       }
       return error.response.headers;
     });
 }
 
+function axiosWorkaround<T extends HttpRequestConfig>(
+  error: AxiosError,
+  axiosConfig: Partial<T>,
+  destination: Destination | DestinationNameAndJwt
+) {
+  if (error.request._isRedirect) {
+    return makeCsrfRequest(destination, {
+      ...axiosConfig,
+      url: error.request._options.path
```

Comment:
Is there a API to get the path? Accessing `_options` is potentially dangerous because this variable is not part of the public API of the request object.

Reply:
Perhaps a check with good logs statement in case the path is not present.

Reply:
will do

Reply:
I added a log statement. I don't think that there is another api for the request, but for extra safety I added a check that the path exists.